File: BC-W4/Army/Spells/WindVortex/WindVortex.cpp
#include "WindVortex.h"

WindVortex::WindVortex(int actionPoints, int manaCost, const std::string& spellName) : Spell(actionPoints, manaCost, spellName) {};
WindVortex::~WindVortex() {};

void WindVortex::action(Unit* enemy) {
    enemy->takeMagicDamage(this->getActionPoints());
};

void WindVortex::heal(Unit* unit) {
    throw InvalidHealException();
};

File: BC-W4/ArmyJava/src/spells/WindVortex.java
package spells;

import exceptions.InvalidCastException;
import units.Unit;

public class WindVortex extends Spell {
    public WindVortex() {
        super("WindVortex", 30, 40, true);
    }

    public WindVortex(int actionPoints) {
        super("WindVortex", 30, actionPoints, true);
    }

    public void action(Unit enemy) throws InvalidCastException {
        enemy.takeMagicDamage(this.actionPoints);
    }

}
